zql/docs/grouping/README.md

```diff
@@ -0,0 +1,181 @@
+# Grouping
+
+All [aggregate functions](../aggregate-functions/README.md) may be invoked with
+one or more _grouping_ options that define the batches of events on which they
+operate. If explicit grouping is not used, an aggregate function will operate
+over all events in the input stream.
+
+Below you will find details regarding the available grouping mechanisms and
+tips for their effective use.
+
+- [Time Grouping - `every`](#time-grouping---every)
+- [Value Grouping - `by`](#value-grouping---by)
+- [Note: Undefined Order](#note-undefined-order)
+
+# Time Grouping - `every`
+
+To create batches of events that are close together in time, specify
+`every <duration>` before invoking your aggregate function(s).
+
+The `<duration>` may be expressed in any of the following units of time. A
+numeric value may also precede the unit specification, in which case any of
+the shorthand variations may also be used.
+
+| **Unit**  | **Plural shorthand (optional)**   |
+|-----------|-----------------------------------|
+| `second`  | `seconds`, `secs`, `sec`, `s`     |
+| `minute`  | `minutes`, `mins`, `min`, `m`     |
+| `hour`    | `hours`, `hrs`, `hr`, `h`         |
+| `day`     | `days`, `d`                       |
+| `week`    | `weeks`, `wks`, `wk`, `w`         |
+
+#### Example #1:
+
+To see the total number of bytes originated across all connections during each
+minute:
+
+```zq-command
+zq -f table 'every minute sum(orig_bytes) | sort -r ts' conn.log.gz
```

Comment:
I think it would be helpful to explain that `every X` is just syntactic sugar for `by Time.trunc(ts, X)`.

Reply:
@henridf and I discussed this suggestion offline and I'm inclined to leave it out for now. Concepts like syntactic sugar (whether we go with that phrasing or try to come up with alternate words) feels a bit deeper than we're trying to go for the audience I have in mind for the current ZQL docs. I expect at some point we may have alternate/supplemental docs for users that come from a programming or data science background, and I suspect that's when it'll become helpful to expose things like this as well as other implementation details. I'll leave this comment "unresolved" in the event that users do happen to come across this closed PR, though.